File: cocache-core/src/main/java/me/ahoo/cache/join/JoinCache.java
package me.ahoo.cache.join;

import me.ahoo.cache.CacheGetter;

/**
 * Join Caching.
 *
 * @author ahoo wang
 */
public interface JoinCache<K1, V1, K2, V2> extends CacheGetter<K1, JoinValue<V1, K2, V2>> {
    
    ExtractJoinKey<V1, K2> getExtractJoinKey();
}

File: cocache-core/src/main/kotlin/me/ahoo/cache/join/JoinCache.kt
package me.ahoo.cache.join

import me.ahoo.cache.CacheGetter

/**
 * Join Caching.
 *
 * @author ahoo wang
 */
interface JoinCache<K1, V1, K2, V2> : CacheGetter<K1, JoinValue<V1, K2, V2>?> {
    val extractJoinKey: ExtractJoinKey<V1, K2>
}
